bgpd/bgp_vty.c

```diff
@@ -8786,33 +9292,461 @@ static void bgp_show_peer_afi_orf_cap(struct vty *vty, struct peer *p,
+						.t_select_deferral != NULL) {
+
+						json_object_int_add(
+						json_timer,
+						"selectionDeferralTimerRemaining",
+						thread_timer_remain_second(
+						peer->bgp
+						->gr_info[afi][safi]
+						.t_select_deferral));
+					}
+
+				} else {
+
+					vty_out(vty, "     Timers:\n");
+
+					vty_out(vty, "%*s", 6, "");
+					vty_out(vty,
+					"Configured Stale Path Time(sec)%*s: %u\n",
+						8, "",
+					peer->bgp->stalepath_time);
+
+					if (peer->t_gr_stale != NULL) {
+						vty_out(vty, "%*s", 6, "");
+						vty_out(vty,
+						"Stale Path Remaining(sec)%*s: %ld\n",
+						14, "",
+						thread_timer_remain_second(
+						peer->t_gr_stale));
+					}
+					/* Display Configured Selection
+					 * Deferral only when when
+					 * Gr mode is enabled.
+					 */
+					if (CHECK_FLAG(peer->flags,
+						PEER_FLAG_GRACEFUL_RESTART)) {
+						vty_out(vty, "%*s", 6, "");
+						vty_out(vty,
+						"Configured Selection Deferral Time(sec): %u\n",
+						peer->bgp->select_defer_time);
+					}
+
+					if (peer->bgp
+						->gr_info[afi][safi]
+						.t_select_deferral != NULL) {
+
+						vty_out(vty, "%*s", 6, "");
+						vty_out(vty,
+						"Selection Deferral Time Remaining(sec) : %ld\n",
+						thread_timer_remain_second(
+						peer->bgp
+						->gr_info[afi][safi]
+						.t_select_deferral));
+					}
+
+				}
+				if (use_json) {
+					json_object_object_add(json_afi_safi,
+							"endOfRibStatus",
+							 json_endofrib_status);
+					json_object_object_add(json_afi_safi,
+							"timers",
+							 json_timer);
+					json_object_object_add(json,
+						get_afi_safi_str(afi, safi, true),
+							 json_afi_safi);
+				}
+			}
+		}
+	}
+}
+
+static void  bgp_show_neighbor_graceful_restart_time(struct vty *vty,
+							struct peer *p,
+							bool use_json,
+							json_object *json)
+{
+	if (use_json) {
+		json_object *json_timer = NULL;
+
+		json_timer = json_object_new_object();
+
+		json_object_int_add(json_timer,
+					"configuredRestartTimer",
+					p->bgp->restart_time);
+
+		json_object_int_add(json_timer,
+					"recivedRestartTimer",
+					p->v_gr_restart);
+
+		if (p->t_gr_restart != NULL) {
+			json_object_int_add(json_timer,
+					"restartTimerRemaining",
+					thread_timer_remain_second(
+					p->t_gr_restart)
+					);
+		}
+
+		json_object_object_add(json, "timers", json_timer);
+	} else {
+
+		vty_out(vty, "    Timers :\n");
+		vty_out(vty,
+			"     Configured Restart Time(sec)  : %u\n",
+				p->bgp->restart_time);
+
+		vty_out(vty,
+			"     Received Restart Time(sec)    : %u\n",
+					p->v_gr_restart);
+		if (p->t_gr_restart != NULL) {
+			vty_out(vty,
+			"     Restart Time Remaining(sec)   : %ld\n",
+				thread_timer_remain_second(
+					p->t_gr_restart));
+		}
+	}
+}
+
+static void bgp_show_peer_gr_status(struct vty *vty, struct peer *p,
+					bool use_json,
+					json_object *json)
+{
+	char buf[SU_ADDRSTRLEN] = {0};
+	char dn_flag[2] = {0};
+	char neighborAddr[INET6_ADDRSTRLEN] = {0};
```

Comment:
can we rename neighborAddr to be snake_case?

Reply:
not sure why it is needed ?